### ISSUE TITLE
Getting Bad Credentials error when passing token input in GHE

### ISSUE THREAD

**Issue by alvaro-ajv** [actions/setup-python#666]
**Description:**
When trying to pass the token input to reduce rate limit error, we are getting "Bad Credentials" error; using previous versions of the action it works, but using the last release we get that error. 

Seems that the bug was introduced on these PR: [443](https://github.com/actions/setup-python/pull/443)
Also the same change from the mentioned PR was tested and reverted in these PRs respectively: [427](https://github.com/actions/setup-python/pull/427) and [437](https://github.com/actions/setup-python/pull/437)

Action version 4.2.0 that is before of the change works fine.

We are trying to use the latest version of the action due to deprecation of "set-output" command as described in the [docs](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/), and as is mentioned in PR [578](https://github.com/actions/setup-python/issues/578), latest version should remove the warning, but we cannot use later versions than 4.2.0 due to above bug.

Bad credentials error:

![image](https://github.com/actions/setup-python/assets/40005341/47bbc20d-230b-4a2f-bc21-46a885ac769d)

"set-ouput" warning:

![image](https://github.com/actions/setup-python/assets/40005341/c767649b-84f7-4ebf-b4a4-dce8f6d606ff)

 

**Action version:**
Latest

**Platform:**
- [x] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [ ] Hosted
- [x] Self-hosted

**Comment by IvanZosimov** [comment]
Hi, @alvaro-ajv 👋 Thanks for the issue, we will take a look!

**Comment by KFo3** [comment]
Similar to Disney's error might be the same bug.  Please correct me if I am wrong.
[ZD Ticket](https://github.zendesk.com/agent/tickets/2169251) 

 Disney just started experiencing this problem after updating GHES to 3.8.3.  They are also utilizing PAT over tokens.  Self hosted runners using  `ubuntu:jammy` 22.04.2

Current runner version: '2.304.0'

```
Run actions/setup-python@v4
Version 3.10 was not found in the local cache
Error: Bad credentials
```

**Comment by priya-kinthali** [comment]
Hello @alvaro-ajv 👋,
Could you kindly verify if the issue related to the rate limit has been resolved and confirm that no other issues have emerged while using the [setup-actions-demo/setup-python@main branch](https://github.com/setup-actions-demo/setup-python).

**Comment by priya-kinthali** [comment]
Hello @alvaro-ajv 👋,
Just a gentle reminder, Could you kindly verify if the issue related to the rate limit has been resolved and confirm that no other issues have emerged while using the [setup-actions-demo/setup-python@main branch](https://github.com/setup-actions-demo/setup-python).

**Comment by priya-kinthali** [comment]
Hello @alvaro-ajv 👋,
As part of the latest updates, the potential improvement to leverage the raw API for retrieving version-manifest has been implemented and released in [v5.1.0](https://github.com/actions/setup-python/releases/tag/v5.1.0). This new approach should mitigate the rate limit exhaustion without requiring an additional GitHub token.
Please update to the latest version and let us know if you encounter any further issues. As this update resolves the reported problem, we will close this issue.
Thank you!
